pydicom/dataset.py

```diff
@@ -329,7 +329,7 @@ class Dataset:
     >>> def recurse(ds):
     ...     for elem in ds:
     ...         if elem.VR == 'SQ':
-    ...             [recurse(item) for item in elem]
+    ...             [recurse(item) for item in elem.value]
```

Comment:
I think this is not needed - shall work without the change, as the iterator will access the value automatically.

Reply:
Ahh cool, okay didn't know that. Cheers :)

Reply:
[I did suggest](https://github.com/pydicom/pydicom/issues/1137#issuecomment-705177711) we remove that, though. I'm not a fan of `DataElement` being iterable

Reply:
I've seen that @scaramallion recommendation has made the v3.0 checklist:

https://github.com/pydicom/pydicom/issues/1232#issue-725877177

Potentially it's worth making the change to the docs now so that new code will be forwards compatible for this pattern?

Reply:
Yeah, I think so

Reply:
I agree that this syntax should be encouraged.